test/util/all_close_f.cpp

```diff
@@ -299,20 +286,33 @@ ::testing::AssertionResult test::all_close_f(const vector<float>& a,
         median_distance = median_sum / 2;
     }
 
-    ar_fail << "passing criteria: " << (mantissa_bits - tolerance_bits) << " mantissa bits ("
-            << mantissa_bits << " mantissa bits w/ " << tolerance_bits << " tolerance bits)\n";
-    ar_fail << std::setprecision(std::numeric_limits<long double>::digits10 + 1)
-            << "tightest match:   " << matching_mantissa_bits(min_distance) << " mantissa bits ("
-            << a[min_distance_index] << " vs " << b[min_distance_index] << " at ["
-            << min_distance_index << "])\n";
-    ar_fail << std::setprecision(std::numeric_limits<long double>::digits10 + 1)
-            << "loosest match:    " << matching_mantissa_bits(max_distance) << " mantissa bits ("
-            << a[max_distance_index] << " vs " << b[max_distance_index] << " at ["
-            << max_distance_index << "])\n";
-    ar_fail << "median match:     " << matching_mantissa_bits(median_distance)
-            << " mantissa bits\n";
-
-    return rc ? ::testing::AssertionSuccess() : ar_fail;
+#ifdef NGRAPH_GTEST_INFO
+    if (rc)
+    {
+        // Short unobtrusive message when passing
+        std::cout << "[   INFO   ] Verifying match of >= " << (mantissa_bits - tolerance_bits)
+                  << " mantissa bits (" << mantissa_bits << " bits precision - " << tolerance_bits
+                  << " tolerance). Loosest match found is " << matching_mantissa_bits(max_distance)
+                  << " mantissa bits.\n";
+    }
+#endif
+
+    msg << "passing criteria: " << (mantissa_bits - tolerance_bits) << " mantissa bits ("
+        << mantissa_bits << " mantissa bits w/ " << tolerance_bits << " tolerance bits)\n";
+    msg << std::setprecision(std::numeric_limits<long double>::digits10 + 1)
+        << "tightest match:   " << matching_mantissa_bits(min_distance) << " mantissa bits ("
+        << a[min_distance_index] << " vs " << b[min_distance_index] << " at [" << min_distance_index
+        << "])\n";
+    msg << std::setprecision(std::numeric_limits<long double>::digits10 + 1)
+        << "loosest match:    " << matching_mantissa_bits(max_distance) << " mantissa bits ("
+        << a[max_distance_index] << " vs " << b[max_distance_index] << " at [" << max_distance_index
+        << "])\n";
+    msg << "median match:     " << matching_mantissa_bits(median_distance) << " mantissa bits\n";
+
+    ::testing::AssertionResult res =
+        rc ? ::testing::AssertionSuccess() : ::testing::AssertionFailure();
+    res << msg.str();
```

Comment:
Does success print out these messages? If so I vote no, if not then why the change?

Reply:
Define "success". Success depends on what result is expected in a given test. For most tests this is ASSERT_TRUE and EXPECT_TRUE, which will be silent when rc is true and res gets ::testing::AssertionSuccess().

But ASSERT_FALSE and EXPECT_FALSE will print a message when rc is true and res gets ::testing::AssertionSuccess(). This actually can happen in the unit testing to verify the correct functioning of all_close_f which does EXPECT_TRUE for cases where all_close_f is expected to return true/AssertionSuccess and EXPECT_FALSE for cases where all_close_f is expected to return false/AssertionFailure.

If we're testing a condition where all_close_f is supposed to fail / return false/AssertionFailure (by using EXPECT_FALSE) and all_close_f instead returns true/AssertionSuccess (because for example a change has broken all_close_f) then the message will be printed.